src/ameba.ts

```diff
@@ -29,22 +30,29 @@ export class Ameba {
         this.config = getConfig();
     }
 
-    public execute(document: TextDocument, virtual: boolean = false): void {
-        if (!isCrystalDocument(document)) return;
-        if (isDocumentVirtual(document) && !virtual) return;
+    public execute(document: TextDocument | WorkspaceFolder, virtual: boolean = false): void {
+        const isWorkspace = !('languageId' in document);
+        if (isWorkspace) virtual = false;
+
+        if (!isWorkspace) {
+            if (!isCrystalDocument(document)) return;
+            if (isDocumentVirtual(document) && !virtual) return;
+        }
 
         const dir = (workspace.getWorkspaceFolder(document.uri) ?? noWorkspaceFolder(document.uri)).uri.fsPath;
 
         const args = [this.config.command, '--format', 'json'];
 
-        if (!virtual) {
-            args.push(document.fileName)
-        } else {
-            // Disabling these as they're common when typing
-            args.push('--except', 'Lint/Formatting,Layout/TrailingBlankLines,Layout/TrailingWhitespace');
+        if (!isWorkspace) {
```

Comment:
You're not passing the folder path anywhere when you're dealing with workspace instead of a file.

Reply:
I don't need to, as running `ameba` from the root of the workspace will run it on that workspace. Passing the workspace path will instead have the downside of ignoring all the files in the `.ameba.yml` Excluded option.